odbc/src/api.rs

```diff
@@ -340,14 +380,17 @@ pub extern "C" fn SQLDescribeColW(
 
 #[no_mangle]
 pub extern "C" fn SQLDescribeParam(
-    _statement_handle: HStmt,
+    statement_handle: HStmt,
     _parameter_number: USmallInt,
     _data_type_ptr: *mut SqlDataType,
     _parameter_size_ptr: *mut ULen,
     _decimal_digits_ptr: *mut SmallInt,
     _nullable_ptr: *mut SmallInt,
 ) -> SqlReturn {
-    unimplemented!()
+    let handle = MongoHandleRef::from(statement_handle);
+    handle.clear_diagnostics();
+    handle.add_diag_info(ODBCError::Unimplemented("SQLDescribeParam"));
+    SqlReturn::ERROR
 }
 
 #[no_mangle]
```

Comment:
I just realized that some functions in api.rs still return unimplemented!(). Was there a reason why you skipped some functions? Or did you simply miss updating the code for them? All functions should return either SQL_ERROR or SQL_SUCCESS (or SUCCESS_WITH_INFO) for the driver to behave correctly. Keeping unimplemented!() means the driver will crash when the function is called.
Could you do another pass at it?

Now, on the same kind of topic, do you think a sql_unimplemented macro or helper function, which would take a MongoHandle and function name and return SQL_ERROR would help will the repetitive code of clearing diagnostics, add a new unimplemented error message and returning SQL_ERROR?

Reply:
This ticket is just supposed to cover unsupported functions, which is why I left those functions as-is. My impression was that we would remove `unimplemented!()` from supported functions as we implement them.

> Keeping unimplemented!() means the driver will crash when the function is called

Will we need to call unimplemented functions before they get properly implemented?

Re: a helper function, I've abstracted away the redundant code via a new function called `fn unsupported_function`

Reply:
Got it now! It makes sense :)
And I like the `unsupported_function` helper 👍